monix-catnap-micrometer/src/main/scala/com/avast/sst/monix/catnap/micrometer/MicrometerCircuitBreakerMetrics.scala

```diff
@@ -0,0 +1,35 @@
+package com.avast.sst.monix.catnap.micrometer
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import cats.effect.Sync
+import com.avast.sst.monix.catnap.CircuitBreakerMetrics
+import com.avast.sst.monix.catnap.CircuitBreakerMetrics.State
+import com.avast.sst.monix.catnap.CircuitBreakerMetrics.State._
+import io.micrometer.core.instrument.MeterRegistry
+
+class MicrometerCircuitBreakerMetrics[F[_]: Sync](name: String, meterRegistry: MeterRegistry) extends CircuitBreakerMetrics[F] {
```

Comment:
shouldn't this class be private?

Reply:
I see that you use it in `Http4sClientCircuitBreakerModule`. But I don't you (or any other user) care that it's `MicrometerCircuitBreakerMetrics` specifically, I think they just care that it's an `CircuitBreakerMetrics` instance.

That leads me to think that `MicrometerCircuitBreakerMetrics` being public is an implementation detail leaking out and that the class should be private. You can then create an instance with:
```scala
object MicrometerCircuitBreakerMetrics {
  def make[F[_]: Sync](name: String, meterRegistry: MeterRegistry): CircuitBreakerMetrics[F] = new MicrometerCircuitBreakerMetrics(name, meterRegistry)
}
```

Reply:
I was thinking of adding the module but I decided not to because most people probably won't use it directly but through the other interop modules. However I did create the module I would leave the class public anyway because you want to return the most specific think in case someone wanted to instantiate it for some specific reason. Again, initialization modules are not an abstraction of any kind and are not really supposed to hide stuff, they are here to simplify the initialization and make it consistent throughout the whole program..

Reply:
> you want to return the most specific thing

Quite the opposite. We need to hide implementation details, and expose only "interfaces". Not doing that would be a bad API design. Our users then would have to worry about programming against the API, not implementation. Let's not pass that burden onto them, let's not give them that pseudo-choice to begin with.

(it would also bind our hand with future development of the library)

Reply:
I agree with your opinions generally but this is a different case. I think we've already discussed this once. If I were providing some library feature I would of course provide an interface however this is not a library providing feature but only initialization helper (basically). I want to return the most specific thing because I don't know how the user wants to use that thing. Of course the user will most probably use it "under the interface".

Reply:
We've had long personal discussion about this. I plan to do some changes.